File: backend/app/schemas/downtime.py
"""
Schemas: Downtime
"""
from pydantic import BaseModel
from datetime import datetime


class ActiveDowntimeCreate(BaseModel):
    machine_code: str
    reason: str
    category: str
    subcategory: str | None = None
    operator_name: str | None = None
    shift: str | None = None
    is_planned: bool = False
    notes: str | None = None


class ActiveDowntimeRead(BaseModel):
    id: int
    machine_code: str
    reason: str
    category: str
    subcategory: str | None = None
    operator_name: str | None = None
    shift: str | None = None
    start_time: datetime
    is_planned: bool
    notes: str | None = None

    model_config = {"from_attributes": True}


class DowntimeHistoryRead(BaseModel):
    id: int
    machine_code: str
    reason: str
    category: str
    subcategory: str | None = None
    operator_name: str | None = None
    shift: str | None = None
    start_time: datetime
    end_time: datetime
    duration_minutes: float
    is_planned: bool
    notes: str | None = None
    resolved_by: str | None = None

    model_config = {"from_attributes": True}

File: synchro-mes/frontend/src/types/downtime.ts
// ── Downtime ─────────────────────────────────────────────────

export interface ActiveDowntime {
  id: number;
  machine_code: string;
  reason: string;
  category: string;
  subcategory: string | null;
  operator_name: string | null;
  shift: string | null;
  start_time: string;
  is_planned: boolean;
  notes: string | null;
}

export interface ActiveDowntimeCreate {
  machine_code: string;
  reason: string;
  category: string;
  subcategory?: string;
  operator_name?: string;
  shift?: string;
  is_planned?: boolean;
  notes?: string;
}

export interface DowntimeHistory {
  id: number;
  machine_code: string;
  reason: string;
  category: string;
  subcategory: string | null;
  operator_name: string | null;
  shift: string | null;
  start_time: string;
  end_time: string;
  duration_minutes: number;
  is_planned: boolean;
  notes: string | null;
  resolved_by: string | null;
}
